### PR TITLE
CRB-256/eager publishing

### PR DESCRIPTION
**Pull request by CAGS295**

### CODE REVIEW

**Review by atodorov — CHANGES_REQUESTED** [review]
Please look at the Clippy log.

File: src/futures/update_stream.rs [hunk 1]
```diff
@@ -257,4 +261,22 @@ mod tests {
...
+  #[allow(dead_code)]
+  fn publisher_timer_resets_after_timed_and_eager_publishing_fail() {}
+
+  /// publishing timer will activate just as on_block_commit is being executed.
+  /// Unluckily the tasks executed go in the following order,
+  /// on_block_commit -> publishing_timer -> updater -> committer
+  /// The miner is reset, publishing will be tried, a consensus won't be ready, the committer reesets publishing.
+  /// Relies on the consensus not finding a valid solution on time.
+  #[allow(dead_code)]
+  fn publisher_timer_slightly_ahead_of_committer_event() {}
```
Comment (nathanwhit) [inline src/futures/update_stream.rs:281]: What's the purpose of these? Are they basically just TODOs?

**Review by nathanwhit — COMMENTED** [review]
I'm a bit confused, does this PR add anything on #11? I don't see any changes here that aren't in #11. I guess maybe #11 was based off of this PR?

File: src/futures/update_stream.rs (hunk 1 above)
Comment (CAGS295) [reply src/futures/update_stream.rs:281]: these are scenarios and edge-cases that I considered while implementing the feature but it was very hard to actually replicate the scenario. I wanted to leave some record and they are also valid TODO tests.

**Review by atodorov — APPROVED** [review]
Previous clippy issues were resolved. I've used the commits from this PR in my recent testing and also as part of the 2.0 testnet deployment. So far testing was PASS so approving this PR.

**Review by nathanwhit — APPROVED** [review]
Looks good to me
